### PR TITLE
Doc: init KV1 connector

### PR DESCRIPTION
**Pull request by papailio**

### CODE REVIEW

File: src/documentation/kv1_read.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
+# KV1 reading specification
+## Introduction
+This document describes how a KV1 feed is read in Navitia Transit model (NTM) and transformed into a [NTFS feed](https://github.com/CanalTP/navitia/blob/dev/documentation/ntfs/ntfs_fr.md).
+
+For the sake of simplicity, the follownig specification describes only those NTFS fields that are specified in the source data (e.g. the `network_url` is not specified and therefore not detailed.)
```
Comment (woshilapin) [inline src/documentation/kv1_read.md]: `follownig` -> `following`

File: src/documentation/kv1_read.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+stop_name | USRSTOPXXX.TMI | *Name* | 
+location_type |  |  | Fixed value `0`.
+stop_lat | POINTXXXXX.TMI | *LocationX_EW* | See bellow for the link of a Point with a UserStop.
+stop_lon | POINTXXXXX.TMI | *LocationY_NS* | See bellow for the link of a Point with a UserStop.
+parent_station | USRSTOPXXX.TMI | *UserStopAreaCode* | 
+platform_code | USRSTOPXXX.TMI | *StopSideCode* | 
+
+**Definining the coordinate of a stop_point :**
+
+The latitude/longitude of a stop_point correspond to the fields *LocationX_EW*, *LocationY_NS* in the file POINTXXXXX.TMI of the point whose *PointCode* matches *UserstopCode* and *PointType* equals the value `SP`. The input coordinate system Amersfoort / RD New (EPSG:28992) should be converted to WGS84 (EPSG:4326).
```
Comment (datanel) [inline src/documentation/kv1_read.md]: corresponds

File: src/documentation/kv1_read.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+stop_lat |  |  | The coordinates of the stop_area are computed as the barycentre of all the associated stop_points.
+stop_lon |  |  | The coordinates of the stop_area are computed as the barycentre of all the associated stop_points.
+location_type |  |  | Fixed value `1`.
+
+### lines.txt
+NTFS field | KV1 file | KV1 field | Mapping rule/Comment
+--- | --- | --- | ---
+line_id | LINEXXXXXX.TMI | *LinePlanningNumber* | This field is prefixed.
+line_code | LINEXXXXXX.TMI | *LinePublicNumber* | 
+line_name |  |  | This field is computed using the the name of the assoicated Route in the forward direction (identical to forward_line_name).
```
Comment (datanel) [inline src/documentation/kv1_read.md]: associated

File: src/documentation/kv1_read.md [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+forward_line_name |  |  | This field is computed using the the name of the assoicated Route in the forward direction.
```
Comment (datanel) [inline src/documentation/kv1_read.md]: associated

File: src/documentation/kv1_read.md [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+forward_direction |  |  | This field is computed using the last stop_area of the associated Route in the forward direction.
```
Comment (datanel) [inline src/documentation/kv1_read.md]: associated

File: src/documentation/kv1_read.md [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+backward_line_name |  |  | This field is computed using the the name of the assoicated Route in the backward direction.
```
Comment (datanel) [inline src/documentation/kv1_read.md]: associated

File: src/documentation/kv1_read.md [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+backward_direction |  |  | This field is computed using the last stop_area of the associated Route in the backward direction.
```
Comment (datanel) [inline src/documentation/kv1_read.md]: associated

File: src/documentation/kv1_read.md [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+exception_type |  |  | Fixed value `1`.
+
+### trips.txt
+NTFS field | KV1 file | KV1 field | Mapping rule/Comment
+--- | --- | --- | ---
+route_id | JOPAXXXXXX.TMI | *LinePlanningNumber*, *Direction* | This field is prefixed. Link to the files [routes.txt](#routestxt). Concatenation of *LinePlanningNumber* and *Direction* separated by a `:`. Ex: "\<prefix\>:2029:2"
+service_id | PUJOPASSXX.TMI | *OrganizationalUnitCode*, *ScheduleCode*, *ScheduleTypeCode* | This field is prefixed. Link to the file [calendar_dates.txt](#calendardatestxt). Concatenation of the 3 specified fields separated by a ':'. Ex: "\<prefix\>:2029:1:1"
+trip_id | JOPAXXXXXX.TMI, PUJOPASSXX.TMI | *LinePlanningNumber*, *JourneyPatternCode*, *JourneyNumber* | This field is prefixed. Concatenation of the 3 specified fields separated by a `:`. Ex: "\<prefix\>:2029:9001:23366"
+company_id | JOPAXXXXXX.TMI | *DataOwnerCode* | This field is prefixed. Link to the file [companies.txt](#companiestxt).
+physical_mode_id |  |  | This field is not prefixed. Link to the file [physical_modes.txt](#physical_modestxt). It is computed using the type of transportation specified for the associated line of the trip.
```
Comment (datanel) [inline src/documentation/kv1_read.md]: can you add KV1 file= `LINEXXXXXX.TMI`  and KV1 field = `TransportType` please?

File: src/documentation/kv1_read.md [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+BUS | Bus | Bus | Bus | Bus
+TRAIN | Train | Train | Train | Train
+METRO | Metro | Metro | Metro | Metro
+TRAM | Tramway | Tramway | Tramway | Tramway
+BOAT | Ferry | Ferry | Ferry | Ferry
+
+### stop_times.txt
+NTFS field | KV1 file | KV1 field | Mapping rule/Comment
+--- | --- | --- | ---
+trip_id | PUJOPASSXX.TMI | *LinePlanningNumber*, *JourneyPatternCode*, *JourneyNumber* | This field is prefixed. Link to the file [trips.txt](#tripstxt). Concatenation of the 3 specified fields separated by a `:`. Ex: "\<prefix\>:2029:9001:23366"
```
Comment (woshilapin) [inline src/documentation/kv1_read.md]:
```suggestion
trip_id | PUJOPASSXX.TMI | *LinePlanningNumber*, *JourneyPatternCode*, *JourneyNumber* | This field is prefixed. Link to the file [trips.txt](#tripstxt). Concatenation of the 3 specified fields separated by a `:`. Ex: `<prefix>:2029:9001:23366`
```

File: src/documentation/kv1_read.md [hunk 10]
```diff
@@ -0,0 +1,125 @@
...
+stop_sequence | PUJOPASSXX.TMI | *StopOrder* | 
+
+### comments.txt
+NTFS field | KV1 file | KV1 field | Mapping rule/Comment
+--- | --- | --- | ---
+comment_id | NOTICEXXXX.TMI | *Notice coder* | This field is prefixed.
+comment_name | NOTICEXXXX.TMI | *Notice (content)* | 
+
+### comment_links.txt
+Only coments on trips (Object field specified with `PUJOPASS`) will be handeled.
```
Comment (woshilapin) [inline src/documentation/kv1_read.md]:
```suggestion
Only comments on trips (Object's field specified with `PUJOPASS`) will be handled.
```

File: src/documentation/kv1_read.md (hunk 5 above)
Comment (nberard) [inline src/documentation/kv1_read.md]: should forward_direction and backward_direction use different stop area?

File: src/documentation/kv1_read.md [hunk 11]
```diff
@@ -0,0 +1,125 @@
...
+
+**Definining the coordinate of a stop_point :**
+
+The latitude/longitude of a stop_point correspond to the fields *LocationX_EW*, *LocationY_NS* in the file POINTXXXXX.TMI of the point whose *PointCode* matches *UserstopCode* and *PointType* equals the value `SP`. The input coordinate system Amersfoort / RD New (EPSG:28992) should be converted to WGS84 (EPSG:4326).
+
+**For stop_areas**
+
+NTFS field | KV1 file | KV1 field | Mapping rule/Comment
+--- | --- | --- | ---
+stop_id | USRSTARXXX.TMI | *UserstopCode* | This field is prefixed.
```
Comment (datanel) [inline src/documentation/kv1_read.md]:
```suggestion
stop_id | USRSTARXXX.TMI | *UserStopAreaCode* | This field is prefixed.
```

File: src/documentation/kv1_read.md [hunk 12]
```diff
@@ -0,0 +1,125 @@
...
+route_name | JOPAXXXXXX.TMI |  | "[most frequent first stop of contained trips] - [most frequent last stop of contained trips]"
+direction_type | JOPAXXXXXX.TMI | *Direction* | `forward` value when *Direction* is `1` or `A`. `backward` in all other cases.
+line_id | JOPAXXXXXX.TMI | *LinePlanningNumber* | This field is prefixed. Link to the file [lines.txt](#linestxt).
+destination_id |  |  | This field is computed using the most frequent last stop_area of contained trips.
+
+### calendar_dates.txt
+NTFS field | KV1 file | KV1 field | Mapping rule/Comment
+--- | --- | --- | ---
+service_id | OPERDAYXXX.TMI | *OrganizationalUnitCode*, *ScheduleCode*, *ScheduleTypeCode* | This field is prefixed. Concatenation of the 3 specified fields separated by a ':'. Ex: "\<prefix\>:2029:1:1"
+date | OPERDAYXXX.TMI |  | Service date to be transformed into the YYYYMMDD format.
```
Comment (datanel) [inline src/documentation/kv1_read.md]:
```suggestion
date | OPERDAYXXX.TMI | ValidDate | Service date to be transformed into the YYYYMMDD format.
```

File: src/documentation/kv1_read.md (hunk 8 above)
Comment (datanel) [inline src/documentation/kv1_read.md]:
```suggestion
physical_mode_id | LINEXXXXXX.TMI | TransportType | This field is not prefixed. Link to the file [physical_modes.txt](#physical_modestxt). It is computed using the type of transportation specified for the associated line of the trip.
```

File: src/documentation/kv1_read.md [hunk 13]
```diff
@@ -0,0 +1,125 @@
...
+exception_type |  |  | Fixed value `1`.
+
+### trips.txt
+NTFS field | KV1 file | KV1 field | Mapping rule/Comment
+--- | --- | --- | ---
+route_id | JOPAXXXXXX.TMI | *LinePlanningNumber*, *Direction* | This field is prefixed. Link to the files [routes.txt](#routestxt). Concatenation of *LinePlanningNumber* and *Direction* separated by a `:`. Ex: "\<prefix\>:2029:2"
```
Comment (datanel) [inline src/documentation/kv1_read.md]:
```suggestion
route_id | JOPAXXXXXX.TMI | *LinePlanningNumber*, *Direction* | Linked by *JourneyPatternCode* in `PUJOPASSXX.TMI` and `JOPAXXXXXX.TMI`. This field is prefixed. Link to the files [routes.txt](#routestxt). Concatenation of *LinePlanningNumber* and *Direction* separated by a `:`. Ex: "\<prefix\>:2029:2"
```

File: src/documentation/kv1_read.md [hunk 14]
```diff
@@ -0,0 +1,125 @@
...
+
+### stops.txt
+#### For stop_points
+
+NTFS field | KV1 file | KV1 field | Mapping rule/Comment
+--- | --- | --- | ---
+stop_id | USRSTOPXXX.TMI | *UserstopCode* | This field is prefixed.
+stop_name | USRSTOPXXX.TMI | *Name* | 
+location_type |  |  | Fixed value `0`.
+stop_lat | POINTXXXXX.TMI | *LocationX_EW* | See below for the link of a Point with a UserStop.
```
Comment (woshilapin) [inline src/documentation/kv1_read.md]: `stop_lat` should be `LocationY_NS` (ordinate axis).
```suggestion
stop_lat | POINTXXXXX.TMI | *LocationY_NS* | See below for the link of a Point with a UserStop.
```

File: src/documentation/kv1_read.md [hunk 15]
```diff
@@ -0,0 +1,125 @@
...
+stop_lon | POINTXXXXX.TMI | *LocationY_NS* | See below for the link of a Point with a UserStop.
```
Comment (woshilapin) [inline src/documentation/kv1_read.md]: `stop_lon` should be `LocationX_EW` (abscissa axis).
```suggestion
stop_lon | POINTXXXXX.TMI | *LocationX_EW* | See below for the link of a Point with a UserStop.
```

File: src/documentation/kv1_read.md [hunk 16]
```diff
@@ -0,0 +1,125 @@
...
+--- | --- | --- | ---
+stop_id | USRSTOPXXX.TMI | *UserstopCode* | This field is prefixed.
+stop_name | USRSTOPXXX.TMI | *Name* | 
+location_type |  |  | Fixed value `0`.
+stop_lat | POINTXXXXX.TMI | *LocationX_EW* | See bellow for the link of a Point with a UserStop.
+stop_lon | POINTXXXXX.TMI | *LocationY_NS* | See bellow for the link of a Point with a UserStop.
+parent_station | USRSTOPXXX.TMI | *UserStopAreaCode* | 
+platform_code | USRSTOPXXX.TMI | *StopSideCode* | 
+
+**Definining the coordinate of a stop_point :**
```
[the remark this answers is not in the document]
Comment (prhod) [reply src/documentation/kv1_read.md]: IMHO, this should not be a title, it's more like a specific point to be  highlighted.

File: src/documentation/kv1_read.md [hunk 17]
```diff
@@ -0,0 +1,125 @@
...
+service_id | OPERDAYXXX.TMI | *OrganizationalUnitCode*, *ScheduleCode*, *ScheduleTypeCode* | This field is prefixed. Concatenation of the 3 specified fields separated by a ':'. Ex: "\<prefix\>:2029:1:1"
+date | OPERDAYXXX.TMI |  | Service date to be transformed into the YYYYMMDD format.
+exception_type |  |  | Fixed value `1`.
+
+### trips.txt
+NTFS field | KV1 file | KV1 field | Mapping rule/Comment
+--- | --- | --- | ---
+route_id | JOPAXXXXXX.TMI | *LinePlanningNumber*, *Direction* | This field is prefixed. Link to the files [routes.txt](#routestxt). Concatenation of *LinePlanningNumber* and *Direction* separated by a `:`. Ex: "\<prefix\>:2029:2"
+service_id | PUJOPASSXX.TMI | *OrganizationalUnitCode*, *ScheduleCode*, *ScheduleTypeCode* | This field is prefixed. Link to the file [calendar_dates.txt](#calendardatestxt). Concatenation of the 3 specified fields separated by a ':'. Ex: "\<prefix\>:2029:1:1"
+trip_id | JOPAXXXXXX.TMI, PUJOPASSXX.TMI | *LinePlanningNumber*, *JourneyPatternCode*, *JourneyNumber* | This field is prefixed. Concatenation of the 3 specified fields separated by a `:`. Ex: "\<prefix\>:2029:9001:23366"
```
[the remark this answers is not in the document]
Comment (prhod) [reply src/documentation/kv1_read.md]: the two files are required to grant access to the 3 fields.

File: src/documentation/kv1_read.md (hunk 16 above)
[the remark this answers is not in the document]
Comment (woshilapin) [reply src/documentation/kv1_read.md]: But do not forget to fix the text then `Definining` -> `Defining`.

File: src/documentation/kv1_read.md (hunk 13 above)
Comment (prhod) [reply src/documentation/kv1_read.md]: I don't understand this comment. The `route_id` is only defined by *LinePlanningNumber* and *Direction*. Can you explain me the need for this addition ?

Comment (datanel) [reply src/documentation/kv1_read.md]: PUJOPASSXX.TMI does not have the field `Direction`

Comment (datanel) [reply src/documentation/kv1_read.md]: trips are in `PUJOPASSXX.TMI` and I need to get `route_id` from `JOPAXXXXXX.TMI` so I need a link beetween `PUJOPASSXX.TMI` and `JOPAXXXXXX.TMI` which is `JourneyPatternCode`

File: src/documentation/kv1_read.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (prhod) [reply src/documentation/kv1_read.md]: Why do you think deleting the link is better ?

File: src/documentation/kv1_read.md (hunk 13 above)
Comment (prhod) [reply src/documentation/kv1_read.md]: I've included a notice before the mapping array. I think it's better this way

File: src/documentation/kv1_read.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (woshilapin) [reply src/documentation/kv1_read.md]: ~Why did I even do that :astonished: I can't remember but it seems dumb...~ OK, it took me a while but @datanel made some modifications to my original comment. The original was only about removing the underscore in the link part.
```suggestion
physical_mode_id |  |  | This field is not prefixed. Link to the file [physical_modes.txt](#physicalmodestxt). It is computed using the type of transportation specified for the associated line of the trip.
```
But looking at the whole file, I'm not sure where this anchor is supposed to be: I don't see any title named `physical_modes.txt`. That might be the reason why @datanel has removed entirely the link?

**Pull request merged**